ospfd/ospf_flood.c

```diff
@@ -91,7 +91,7 @@ struct external_info *ospf_external_info_check(struct ospf *ospf,
 	p.prefix = lsa->data->id;
 	p.prefixlen = ip_masklen(al->mask);
 
-	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
```

Comment:
Again is it possible to get a bit more explanation in the commit message about the reasoning here?

Reply:
For any non-default route, it's accessing beyond the end of the array 3 lines below, here:

https://github.com/FRRouting/frr/pull/4742/files/ef7c25cade29d2de3e60bc00425cf55b3c42918d..4a16927557cb3ce40eaf49f2ad621ee48158af42#diff-bde063ebefe3d18e5529fc34b8044f26L102

both `->redist` and `->mi_redist` only have `0 <= x < ZEBRA_ROUTE_MAX`, so it's an off by one checking garbage data.